internal/apis/config/controller/validation/validation.go

```diff
@@ -23,71 +23,96 @@ import (
 	"net/url"
 	"strings"
 
-	//utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/apimachinery/pkg/util/sets"
 
 	config "github.com/cert-manager/cert-manager/internal/apis/config/controller"
 	defaults "github.com/cert-manager/cert-manager/internal/apis/config/controller/v1alpha1"
 )
 
-func ValidateControllerConfiguration(o *config.ControllerConfiguration) error {
-	if len(o.IngressShimConfig.DefaultIssuerKind) == 0 {
-		return errors.New("the --default-issuer-kind flag must not be empty")
```

Comment:
It looks like the original validation only returned one error at a time.
Thanks for fixing that.